### PR TITLE
Complete Code-Challenges: Animal Shelter

### PR DESCRIPTION
**Pull request by Beers15**
- [x] Shelter allows the dequeuing of one animal type mixed in with others without changing the FIFO order
- [x] Shelter allows the enqueuing and dequeuing of either a dog or cat
- [x] The dequeue method returns null if the user enters an animal that is not a cat or dog when dequeueing
- [x] Shelter does not allow the dequeuing of animals if there are no animals in it
- [x] Dequeue method returns null if a valid animal type if preferred, but no animals of that type are in the shelter

### CODE REVIEW

**Comment by Beers15** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Picture of whiteboard
     - [x] Link to code
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)

File: javascript/code-challenges/stack-queue-animal-shelter/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    Shelter allows the dequeuing of one animal type mixed in with others without changing the FIFO order
+    Shelter allows the enqueuing and dequeuing of either a dog or cat
+    The dequeue method returns null if the user enters an animal that is not a cat or dog when dequeueing
+    Shelter does not allow the dequeuing of animals if there are no animals in it
+    Dequeue method returns null if a valid animal type if preferred, but no animals of that type are in the shelter
+
+## Approach & Efficiency
+<!-- What approach did you take? Why? What is the Big O space/time for this approach? -->
+My approach was to find what combination of stacks and queues would work best in terms of efficiency. I ended up implementing this solution with 2 queues. The challenge here was maintaining the FIFO order when the first item wasn't the correct animal. I ended up moving all the data to a 2nd queue when this happened to maintain order integrity which was still relatively efficient.
+
```
Comment (daneng1) [inline javascript/code-challenges/stack-queue-animal-shelter/README.md:26]: I think two queues would be the best approach however it might have been easier to have one queue for dogs and one for cats. Then when there's an adoption request for a dog, you could just dequeue from that queue. But overall, nice work!

**Review by daneng1 — COMMENTED** [review]
Nice work!

File: javascript/code-challenges/stack-queue-animal-shelter/README.md (hunk 1 above)
Comment (Beers15) [reply javascript/code-challenges/stack-queue-animal-shelter/README.md:26]: That's a much better approach, thanks for the tip!
